File: Math challenge/Challanges/Multiplication.m
//
//  Multiplication.m
//  Math challenge
//
//  Created by Lukas Bischof on 09.01.18.
//  Copyright © 2018 Lukas Bischof. All rights reserved.
//

#import "Multiplication.h"

@implementation Multiplication

- (Challange *)next {
    NSInteger first = randr(2, 9);
    NSInteger second = randr(2, 9);
    
    return [[Challange alloc] initWithText:SWF(@"%ld * %ld", (long)first, (long)second) andAnswer:first * second];
}

- (NSString *)challangeDescription
{
    return @"Multiplikation (1-10)";
}

@end

File: Math challenge/challenges/Multiplication.swift
//
// Created by Lukas Bischof on 25.10.19.
// Copyright (c) 2019 Lukas Bischof. All rights reserved.
//

import Foundation

class Multiplication: MathChallengeCategory {
  let challengeDescription: String = "Multiplikation (1-10)"

  func next() -> Challenge {
    let first = randomInt(2, 9)
    let second = randomInt(2, 9)

    return Challenge(text: "\(first) * \(second)", andAnswer: first * second)
  }
}
